R/eval.R

```diff
@@ -1,36 +1,67 @@
 #' Evaluate an expression in an environment
 #'
-#' `eval_bare()` is a lightweight version of the base function
-#' [base::eval()]. It does not accept supplementary data, but it is
-#' more efficient and does not clutter the evaluation stack.
-#' Technically, `eval_bare()` is a simple wrapper around the C
-#' function `Rf_eval()`.
-#'
-#' `base::eval()` inserts two call frames in the stack, the second of
-#' which features the `envir` parameter as frame environment. This may
-#' unnecessarily clutter the evaluation stack and it can change
-#' evaluation semantics with stack sensitive functions in the case
-#' where `env` is an evaluation environment of a stack frame (see
-#' [ctxt_stack()]). Since the base function `eval()` creates a new
-#' evaluation context with `env` as frame environment there are
-#' actually two contexts with the same evaluation environment on the
-#' stack when `expr` is evaluated. Thus, any command that looks up
-#' frames on the stack (stack sensitive functions) may find the
-#' parasite frame set up by `eval()` rather than the original frame
-#' targetted by `env`. As a result, code evaluated with `base::eval()`
-#' does not have the property of stack consistency, and stack
-#' sensitive functions like [base::return()], [base::parent.frame()]
-#' may return misleading results.
+#' @description
+#'
+#' `eval_bare()` is a lower-level version of function [base::eval()].
```

Comment:
I think this needs a sentence or two about why you might use it instead of `eval()`. 

Currently the description is probably a bit long, so you could then move the bulleted list to details.